app/services/bulk_upload/sales/year2024/row_parser.rb

```diff
@@ -1300,12 +1301,17 @@ def validate_owning_org_owns_stock
   end
 
   def validate_owning_org_permitted
-    if owning_organisation && !bulk_upload.user.organisation.affiliated_stock_owners.include?(owning_organisation)
-      block_log_creation!
+    return unless owning_organisation
+    return if bulk_upload_organisation.affiliated_stock_owners.include?(owning_organisation)
 
-      if errors[:field_1].blank?
-        errors.add(:field_1, "You do not have permission to add logs for this owning organisation", category: :setup)
-      end
+    block_log_creation!
+
+    return if errors[:field_1].present?
+
+    if bulk_upload.user.support?
+      errors.add(:field_1, "This owning organisation is not affiliated with #{bulk_upload_organisation.name}", category: :setup)
+    else
+      errors.add(:field_1, "You do not have permission to add logs for this owning organisation", category: :setup)
     end
   end
```

Comment:
I now use the bulk upload organisation id attribute to check if the owning organisation is affiliated, since for each bulk upload organisation_id is either the specified org (support) or the user's own org (non-support). Should this fail the appropriate error message is given.